e2e/test/scenarios/question/notebook.cy.spec.js

```diff
@@ -678,9 +678,27 @@ describe("scenarios > question > notebook", { tags: "@slow" }, () => {
     });
   });
 
-  it("should properly render previews (metabase#28726), (metabase#29959)", () => {
-    openOrdersTable({ mode: "notebook" });
-    cy.findByTestId("step-data-0-0").within(() => {
+  it("should properly render previews (metabase#28726, metabase#29959, metabase#40608)", () => {
+    startNewQuestion();
+
+    cy.log(
+      "Preview should not be possible without the source data (metabase#40608)",
+    );
+    cy.findByTestId("step-data-0-0")
+      .as("dataStep")
+      .within(() => {
+        cy.findByText("Pick your starting data").should("exist");
+        cy.icon("play").should("not.be.visible");
+      });
+
+    popover().findByTextEnsureVisible("Raw Data").click();
+    cy.get("@dataStep").icon("play").should("not.be.visible");
+    popover().findByTextEnsureVisible("Orders").click();
+
+    cy.findByTestId("step-filter-0-0").icon("play").should("not.be.visible");
```

Comment:
nit: `getNotebookStep` can be used

Reply:
Great tip! Thank you.
https://github.com/metabase/metabase/pull/40609/commits/133743f0f59f67f072a8039bcad5a4f788b48984